praw/models/listing/generator.py

```diff
@@ -74,7 +76,12 @@ def _next_batch(self):
         if isinstance(self._listing, list):
             self._listing = self._listing[1]  # for submission duplicates
         elif isinstance(self._listing, dict):
-            self._listing = FlairListing(self._reddit, self._listing)
+            if FlairListing.CHILD_ATTRIBUTE in self._listing:
+                self._listing = FlairListing(self._reddit, self._listing)
+            elif ModNoteListing.CHILD_ATTRIBUTE in self._listing:
+                self._listing = ModNoteListing(self._reddit, self._listing)
```

Comment:
Since there is more, could we create a dynamic solution for this? e.g., iterate through a list of classes instead of checking each one individually.

Reply:
I added this, but it's actually more lines of code for just the two of them. Still think it's worth it?